enumerables.rb

```diff
@@ -0,0 +1,135 @@
+module Enumerable
+
+  def my_each
+    return to_enum unless block_given?
+      i = 0
+      while i < self.size
+        yield(self[i])
+        i += 1
+      end
+      return to_enum
+  end
+
+  def my_each_with_index
+      if block_given?
+        ind = 0
+        size.times do
+          yield(self[ind], ind)
+          ind += 1
+        end
+      else
+        return to_enum
```

Comment:
* [ ] Try to returnt his and see.  to_enum(__method__)

Reply:
please explain?